modules/OtJsonUtilities.js

```diff
@@ -111,7 +111,17 @@ class OtJsonUtilities {
             return datasetCopy;
         case '1.1':
             datasetCopy = Utilities.copyObject(dataset);
-            return JSON.parse(Utilities.sortObjectRecursively(datasetCopy));
+            if (datasetCopy.datasetHeader.transpilationInfo) {
```

Comment:
Can you make one method repackDataset and call it where needed, so that we don't have same code repeating 3 times?